src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,336 @@
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> x = first; x != null; x = x.next) {
+                if (object.equals(x.item)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void checkPositionIndex(int index) {
+        if (!isPositionIndex(index)) {
+            throw new IndexOutOfBoundsException("IndexOutOfBoundsException: " + index);
+        }
+    }
+
+    /**
+     * Links e as last element.
+     */
+    void linkLast(T e) {
+        final Node<T> l = last;
+        final Node<T> newNode = new Node<>(l, e, null);
+        last = newNode;
+        if (l == null) {
+            first = newNode;
+        } else {
+            l.next = newNode;
+        }
+        size++;
+        modCount++;
+    }
+
+    /**
+     * Inserts element e before non-null Node succ.
+     */
+    void linkBefore(T e, Node<T> succ) {
+        // assert succ != null;
+        final Node<T> pred = succ.prev;
+        final Node<T> newNode = new Node<>(pred, e, succ);
+        succ.prev = newNode;
+        if (pred == null) {
+            first = newNode;
+        } else {
+            pred.next = newNode;
+        }
+        size++;
+        modCount++;
+    }
+
+
+    /**
+     * Returns the (non-null) Node at the specified element index.
+     */
+    Node<T> node(int index) {
+        // assert isElementIndex(index);
+
+        if (index < (size >> 1)) {
+            Node<T> x = first;
+            for (int i = 0; i < index; i++) {
+                x = x.next;
+            }
+            return x;
+        } else {
+            Node<T> x = last;
+            for (int i = size - 1; i > index; i--) {
+                x = x.prev;
+            }
+            return x;
+        }
     }
+
+    /**
+     * Unlinks non-null node x.
+     */
+    T unlink(Node<T> x) {
+        // assert x != null;
+        final T element = x.item;
+        final Node<T> next = x.next;
+        final Node<T> prev = x.prev;
+
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            x.prev = null;
+        }
+
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            x.next = null;
+        }
+
+        x.item = null;
+        size--;
+        modCount++;
+        return element;
+    }
+
+    /**
+     * Tells if the argument is the index of a valid position for an
+     * iterator or an add operation.
+     */
+    private boolean isPositionIndex(int index) {
+        return index >= 0 && index <= size;
+    }
+
+    private void checkElementIndex(int index) {
+        if (!isElementIndex(index)) {
```

Comment:
```suggestion
        if (!index >= 0 && index < size;) {
```
в таких випадках не потрібно створювати окремий метод для 1 строки коду